lib/flappy_search_bar.dart

```diff
@@ -405,8 +410,15 @@ class _SearchBarState<T> extends State<SearchBar<T>>
                     duration: Duration(milliseconds: _animate ? 1000 : 0),
                     child: AnimatedContainer(
                       duration: Duration(milliseconds: 200),
-                      width:
-                          _animate ? MediaQuery.of(context).size.width * .2 : 0,
+                      /* If the cancellation widget is set to be shown
+                      (or animated) then set the widget width to 
+                      what the user set. If they didn't set anything
+                      set it to the default size. */
+                      width: _animate
+                          ? (widget.cancellationWidgetSpaceWidth != null
+                              ? widget.cancellationWidgetSpaceWidth
+                              : MediaQuery.of(context).size.width * .2)
```

Comment:
I think that this solution is better :) !

But I think we have still one more little issue 
The animation showing up the cancellation widget also reduce the search bar width like this :

```
AnimatedContainer(
                    duration: Duration(milliseconds: 200),
                    width: _animate ? widthMax * .8 : widthMax,
...
```

So, If we use the `cancellationWidgetSpaceWidth` field, we have to replace : 
```
widthMax * .8
```

by 

```
widthMax - (widget.cancellationWidgetSpaceWidth != null
                        ? widget.cancellationWidgetSpaceWidth
                        : MediaQuery.of(context).size.width * .2)
```

And then it would work perfectly :)
I think that 

```
(widget.cancellationWidgetSpaceWidth != null
                        ? widget.cancellationWidgetSpaceWidth
                        : MediaQuery.of(context).size.width * .2)
```

could be put in one little function like `_getCancellationWidth()` that we could easily call at these 2 locations
And then it would be perfectly fine :)

Reply:
I've tried to do that in the below commit. ("Redid everything...")